### ISSUE TITLE
Deprecate SslRequired.EXTERNAL

### ISSUE THREAD

**Issue by shawkins** [keycloak/keycloak#31914]
If someone has the requirement that all external requests should use TLS, the way to configure this is to allow only TLS connections via the Ingress. There is nothing else that Keycloak should need to do. This is my primary reason why we should deprecate this check in Keycloak altogether. 

To explain more about the current logic in Keycloak how the TLS detection works when Keycloak is behind a proxy that terminates (or re-encrypts) the TLS connection: 

* For Keycloak to know that the connection coming from the Ingress was a HTTPS connection on the outside, the Ingress adds a forwarding HTTP header indicating the original protocol. Keycloak then trusts this information and even though the incoming connection is cleartext, it assumes that the connection is secured. While this works, there is no need to check this anyway if the Ingress is configured to accept only TLS connections. People usually configure an Ingress with a HTTP to HTTPS redirection, plus HSTS, again all items that work very well independent of Keycloak. 

While I'm ok to proceed with this PR, we should have a separate issue to deprecate this feature including a discussion around this. Once it will then be removed, it would default to a no-op, which would be "NONE".

_Originally posted by @ahus1 in https://github.com/keycloak/keycloak/issues/30678#issuecomment-2268432772_

**Comment by shawkins** [comment]
Relates to #17126 - which is full removal.

**Comment by stianst** [comment]
I agree with deprecating ssl-required as that's just stupid at the realm level. Don't agree with the rest though.

Keycloak being an IdP should not just trust folks to do the correct thing. It should warn/fail with bad config.

Ssl-required option in realms require a replacement and that should be a security policy applied to the server, where one setting would be TLS required.

**Comment by shawkins** [comment]
It doesn't appear this has much value in remaining open. #17126 can address how we'll evolve away from this.
